src/celengine/timeline.cpp

```diff
@@ -27,9 +27,7 @@ Timeline::~Timeline()
     for (auto phase : phases)
     {
         // Remove the phase from whatever phase tree contains it.
-        phase->getFrameTree()->removeChild(phase);
-
-        phase->release();
+        phase->getFrameTree()->removeChild(phase.get());
```

Comment:
I believe  FrameTree owns its children so it should accept sharedptr not raw.

Reply:
I believe it's done this way in second commit.